### PR TITLE
[IPv6] Add CBMC test for prvHandleListen.

### PR DESCRIPTION
**Pull request by ActoryOu**
[IPv6] Add CBMC test for prvHandleListen.

Description
-----------

- Add CBMC test cases for the function prvHandleListen, including IPv4 and IPv6.

Test Steps
-----------

```
cd test/cbmc/proof/
python3 prepare.py
cd test_case
make
```

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/TCP/prvHandleListen/prvHandleListen_harness.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                          BaseType_t xType,
+                          BaseType_t xProtocol )
+{
+    return safeMalloc( sizeof( FreeRTOS_Socket_t ) );
+}
+
+void harness()
+{
+    size_t xDataLength;
+    FreeRTOS_Socket_t * pxSocket = ( FreeRTOS_Socket_t * ) safeMalloc( sizeof( FreeRTOS_Socket_t ) );
```
Comment (moninom1) [inline test/cbmc/proofs/TCP/prvHandleListen/prvHandleListen_harness.c:58]: For socket please use existing API ensure_FreeRTOS_Socket_t_is_allocated

File: test/cbmc/proofs/TCP/prvHandleListen/Makefile.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+# BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+# http://aws.amazon.com/freertos
+# http://www.FreeRTOS.org
+#
```
Comment (moninom1) [inline test/cbmc/proofs/TCP/prvHandleListen/Makefile.json:27]: Comments needs to be removed

Comment (ActoryOu) [reply test/cbmc/proofs/TCP/prvHandleListen/Makefile.json]: Updated

File: test/cbmc/proofs/TCP/prvHandleListen/prvHandleListen_harness.c (hunk 1 above)
Comment (ActoryOu) [reply test/cbmc/proofs/TCP/prvHandleListen/prvHandleListen_harness.c]: Updated

**Review by moninom1 — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**